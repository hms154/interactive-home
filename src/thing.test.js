import React from 'react';
import Thing from './thing';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

describe('<Thing />', () => {
    it('should have a name when created', () => {
        const tree = renderer.create(
            <Thing name = "light 1" />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    
    })
    it('should display "On" when props.value is true', () => {
        const thing = shallow(
            <Thing name = "light 1" value={true} />
        );

        expect(thing.text()).toBe('light 1: On');
    });
    it('should display "Off" when value props.value is false', () => {
        const thing = shallow(
            <Thing name = "light 1" value={false} />
        );

        expect(thing.text()).toBe('light 1: Off');
    });
})