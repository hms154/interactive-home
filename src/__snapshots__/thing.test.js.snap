exports[`<Thing /> should have a name when created 1`] = `
<p>
  light 1: Off
</p>
`;
